src/App.tsx

```diff
@@ -15,36 +15,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+export const App = () => {
+  const [sortType, setSortType] = useState('default');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const getSortedGoods = () => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (sortType === 'alphabet') {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortType === 'length') {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
+
+  const handleSortAlphabetically = () => {
+    setSortType('alphabet');
+  };
+
+  const handleSortByLength = () => {
+    setSortType('length');
+  };
+
+  const handleReverse = () => {
+    setIsReversed(prevState => !prevState);
+  };
+
+  const handleReset = () => {
+    setSortType('default');
+    setIsReversed(false);
+  };
+
+  const isActive = sortType !== 'default' || isReversed;
+  const sortedGoods = getSortedGoods();
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortType === 'alphabet' ? '' : 'is-light'}`}
+          onClick={handleSortAlphabetically}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${sortType === 'length' ? '' : 'is-light'}`}
+          onClick={handleSortByLength}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-warning ${isReversed ? '' : 'is-light'}`}
```

Comment:
[UX] - Don't forget to mark the active button to explain to the user what action is active; The button for reversing the order does not have an active state when isReversed is true. Consider adding a class that visually indicates it is active.